stub/UVFunctions.php

```diff
@@ -0,0 +1,2007 @@
+ * @param int $readable
+ *
+ * @return UVTty
+ */
+function uv_tty_init(UVLoop $loop, $fd, int $readable)
+{
+}
+
+/**
+ * Gets the current Window size. On success it returns 0.
+ *
+ * @param UVTty $tty
+ * @param int $width
+ * @param int $height
+ *
+ * @return int
+ */
+function uv_tty_get_winsize(UVTty $tty, int &$width, int &$height)
+{
+}
+
+/**
+ * Set the TTY using the specified terminal mode.
+ *
+ * @param UVTty $tty
+ * @param int $mode
+ *
+ * @return int
+ */
+function uv_tty_set_mode(UVTty $tty, int $mode)
+{
+}
+
+/**
+ * To be called when the program exits.
+ *
+ * Resets TTY settings to default values for the next process to take over.
+ *
+ * This function is async signal-safe on Unix platforms but can fail with error code
+ * UV_EBUSY if you call it when execution is inside uv_tty_set_mode().
+ *
+ * @return void
+ */
+function uv_tty_reset_mode()
+{
+}
+
+/**
+ * Gets the current system uptime.
+ *
+ * @return float
+ */
+function uv_uptime()
+{
+}
+
+/**
+ * Returns current free memory size.
+ *
+ * @return int
+ */
+function uv_get_free_memory()
+{
+}
+
+/**
+ * Gets memory information (in bytes).
+ *
+ * @return int
+ */
+function uv_get_total_memory()
+{
+}
+
+/**
+ * Gets address information about the network interfaces on the system.
+ *
+ * An array of count elements is allocated and returned in addresses.
+ * It must be freed by the user, calling uv_free_interface_addresses().
+ *
+ * @return array
+ */
+function uv_interface_addresses()
+{
+}
+
+/**
+ * Change working directory.
+ *
+ * @param string $directory
+ * @return bool
+ */
+function uv_chdir(string $directory)
+{
+}
+
+/**
+ * Get the current address to which the handle is bound.
+ *
+ * Name must point to a valid and big enough chunk of memory,
+ * struct sockaddr_storage is recommended for IPv4 and IPv6 support.
+ *
+ * @param UVTcp $uv_sock
+ *
+ * @return string
+ */
+function uv_tcp_getsockname(UVTcp $uv_sock)
+{
+}
+
+/**
+ * Get the address of the peer connected to the handle.
+ *
+ * Name must point to a valid and big enough chunk of memory,
+ * struct sockaddr_storage is recommended for IPv4 and IPv6 support.
+ *
+ * @param UVTcp $uv_sock
+ *
+ * @return string
+ */
+function uv_tcp_getpeername(UVTcp $uv_sock)
+{
+}
+
+/**
+ * Get the local IP and port of the UDP handle.
+ *
+ * @param UVUdp $uv_sockaddr
+ *
+ * @return string
+ */
+function uv_udp_getsockname(UVUdp $uv_sock)
+{
+}
+
+/**
+ * Gets the resident set size (RSS) for the current process.
+ *
+ * @return int
+ */
+function uv_resident_set_memory()
+{
+}
+
+////////////////////////
+// Not part of `libuv`
+////////////////////////
+
+/**
+ * @param UVLoop|null $uv_loop
+ *
+ * @return void
+ */
+function uv_run_once(UVLoop $uv_loop = null)
+{
+}
+
+/**
+ * Returns current uv type. (this is not libuv function. util for php-uv).
+ *
+ * @param resource $uv uv_handle.
+ *
+ * @return int  should return UV::IS_UV_* constatns. e.g) UV::IS_UV_TCP.
+ */
+function uv_handle_type($uv)
```

Comment:
It's called uv_handle_get_type in libuv. Not sure why the names are different.  ... Maybe I should rename it?

Reply:
yes, i think so, it seems it returns the same thing.